pro/src/screens/IndividualOffer/DetailsScreen/utils.ts

```diff
@@ -84,23 +125,43 @@ type onSubcategoryChangeProps = {
     value: any,
     shouldValidate?: boolean | undefined
   ) => Promise<void | FormikErrors<DetailsFormValues>>
+  subcategoryConditionalFields: string[]
 }
 
 export const onSubcategoryChange = async ({
   newSubCategoryId,
   subcategories,
   setFieldValue,
+  subcategoryConditionalFields,
 }: onSubcategoryChangeProps) => {
   const newSubcategory = subcategories.find(
     (subcategory) => subcategory.id === newSubCategoryId
   )
 
-  const { subcategoryConditionalFields } =
+  const { subcategoryConditionalFields: newSubcategoryConditionalFields } =
     buildSubcategoryConditonalFields(newSubcategory)
   await setFieldValue(
     'subcategoryConditionalFields',
-    subcategoryConditionalFields
+    newSubcategoryConditionalFields
+  )
+
+  if (newSubcategoryConditionalFields === subcategoryConditionalFields) {
```

Comment:
(question) Est-ce que c'est possible que ce soit égal ? J'ai l'impression que `buildSubcategoryConditonalFields` retourne systématiquement une nouvelle référence de liste.

Reply:
je pense que oui parce que ça fonctionne !

Reply:
Je pense que ça fonctionne, mais à mon avis la ligne 148 est jamais `true`!

Dans le test 'should build subcategory fields', quand tu fais `.toStrictEqual` sur deux objets, ça va faire une comparaison de la structure des 2 objets donc ils sont bien égaux, mais dans le composant le js vérifie juste la référence avec `===`.

Aussi en regardant `buildSubcategoryConditonalFields` est-ce que tu as besoin de retourner un objet contenant ton résultat plutôt que la liste directement ? Vu que tu déstructures l'objet à chaque fois